test/jdk/java/nio/channels/DatagramChannel/Loopback.java

```diff
@@ -157,10 +159,15 @@ static void test(ProtocolFamily family, InetAddress group, NetworkInterface ni)
                     } else {
                         sel.selectedKeys().clear();
                         SocketAddress sender = dc.receive(dst);
+                        if (!src.toString().contentEquals(dst.toString())) {
+                            System.out.println("src: " + src + "not equal to dst: " + dst);
+                            dst.clear();
+                            continue;
+                        }
```

Comment:
This is not the correct way to compare two ByteBuffers. ByteBuffer::toString only output things like position, limit, etc.. I suggest using `ByteBuffer::mismatch` instead.
You will have to `flip` src after line 146 though, since all bytes in src will have been read at that point.